### PR TITLE
WireframeGeometry: Prevent duplicate edges.

### LINKED ISSUE #22094: WireframeGeometry generates duplicate edges
`WireframeGeometry` generates duplicate edges for both indexed and non-indexed buffer geometry.

This is true by inspection, as the code does not test for duplicate xyz-coordinates.

A simple non-indexed example is:
```js
const geometry = new THREE.TetrahedronGeometry( 1, 0 ); // 6 unique edges

const wireframe = new THREE.WireframeGeometry( geometry ); // 12 edges
```

`BoxGeometry( 1, 1, 1 )` is a simple indexed example.

Duplicating an edge is not a problem most of the time, but when using dashed lines, it is a problem.

---

EDIT: Actually, instead of "duplicate edges", they should be referred to as "coincident edges": A -> B  =  B -> A.

### PR DESCRIPTION
**Pull request by Mugen87**
Related issue: Fixed #22094.

**Description**

`WireframeGeometry` checks now for unique edges by comparing vertex data.

### CODE REVIEW

**Comment by Mugen87** [comment]
The current check for uniqueness takes way too much time for complex models. I guess we need a hash again...

File: src/geometries/WireframeGeometry.js [hunk 1]
```diff
@@ -121,5 +112,23 @@ class WireframeGeometry extends BufferGeometry {
 
 }
 
+function isUniqueEdge( start, end, edges ) {
+
+	const hash1 = `${start.x},${start.y},${start.z}-${end.x},${end.y},${end.z}`;
+	const hash2 = `${end.x},${end.y},${end.z}-${start.x},${start.y},${start.z}`; // opposing edge
```
Comment (WestLangley) [inline src/geometries/WireframeGeometry.js:118]: nit: // coincident edge

**Comment by WestLangley** [comment]
This is working for me. Thanks!

**Comment by WestLangley** [comment]
@Mugen87  Are you aware of a reason why EdgesGeometry _requires_ a vertex precision test?

**Comment by mrdoob** [comment]
Beautiful~!

**Comment by mrdoob** [comment]
Thanks!

**Comment by Mugen87** [comment]
> Are you aware of a reason why EdgesGeometry requires a vertex precision test?

I guess the original author wanted to honor https://github.com/mrdoob/three.js/issues/22094#issuecomment-875047216 :shrug:.

**Comment by WestLangley** [comment]
`EdgesGeometry's` hard-wired `precisionPoints` breaks small geometries.

Would you be willing to apply your fix to `EdgesGeometry`, too -- removing the rounding?

That way, `WireframeGeometry` and `EdgesGeometry` would  produce similar results.

**Comment by Mugen87** [comment]
Aligning the implementations sounds good. Let me file a PR.

**Pull request merged**